src/session/session_dhandle.c

```diff
@@ -379,9 +380,13 @@ __session_dhandle_sweep(WT_SESSION_IMPL *session)
     TAILQ_FOREACH_SAFE(dhandle_cache, &session->dhandles, q, dhandle_cache_tmp)
     {
         dhandle = dhandle_cache->dhandle;
+        if (dhandle->type == WT_DHANDLE_TYPE_BTREE)
+            WT_WITH_DHANDLE(session, dhandle, evict_bytes = __wt_btree_bytes_evictable(session));
```

Comment:
I'd consider making the local variable a `bool empty_btree`, and here setting `empty_btree = (__wt_btree_bytes_evictable(session) == 0)`.  Then the check later doesn't need to re-check `dhandle->type`.

Reply:
Wow, thats a really good idea! thanks